src/main/java/nextstep/subway/domain/line/LineStation.java

```diff
@@ -0,0 +1,144 @@
+import nextstep.subway.constants.ErrorMessage;
+import nextstep.subway.domain.BaseEntity;
+import nextstep.subway.domain.station.Station;
+import nextstep.subway.domain.station.StationPosition;
+import nextstep.subway.domain.station.StationRegisterStatus;
+import nextstep.subway.domain.station.StationStatus;
+
+@Entity
+public class LineStation extends BaseEntity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "station_id", foreignKey = @ForeignKey(name = "fk_line_station_station"))
+    private Station station;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "pre_station_id", foreignKey = @ForeignKey(name = "fk_line_station_pre_station"))
+    private Station preStation;
+
+    @Column(nullable = false)
+    private int distance;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "line_id", foreignKey = @ForeignKey(name = "fk_line_station_to_line"))
+    private Line line;
+
+    public LineStation() {
+    }
+
+    public LineStation(Station station, Station preStation, int distance, Line line) {
+        this.station = station;
+        this.preStation = preStation;
+        this.distance = distance;
+        this.line = line;
+    }
+
+    public StationStatus checkStationStatus(Station station) {
+        if (station.equals(this.station)) {
+            return new StationStatus(station, this, StationPosition.DOWNSTATION);
+        }
+        if (station.equals(this.preStation)) {
+            return new StationStatus(station, this, StationPosition.UPSTATION);
+        }
+        return null;
+    }
+
+    public int distanceCompare(int distance) {
+        return Integer.compare(this.distance, distance);
+    }
+
+    public void splitLineStation(StationPosition stationPosition, Station interStation, int subtractDistance) {
+        if (stationPosition == StationPosition.UPSTATION) {
+            preStation = interStation;
+        }
+        if (stationPosition == StationPosition.DOWNSTATION) {
+            station = interStation;
+        }
+        distance -= subtractDistance;
+    }
+
+    public void validate(LineStations lineStations) {
+        StationRegisterStatus upStationStatus = lineStations.getStationRegisterStatus(preStation);
+        StationRegisterStatus downStationStatus = lineStations.getStationRegisterStatus(station);
+        checkRegisteredStationExist(upStationStatus, downStationStatus);
+        checkBothStationRegistered(upStationStatus, downStationStatus);
+        upStationStatus.validate(StationPosition.UPSTATION, distance, station);
+        downStationStatus.validate(StationPosition.DOWNSTATION, distance, preStation);
+    }
+
+    private void checkRegisteredStationExist(StationRegisterStatus upStationStatus,
+            StationRegisterStatus downStationStatus) {
+        if (upStationStatus.isEmpty() && downStationStatus.isEmpty()) {
```

Comment:
```suggestion
        if (isStationsEmpty()) {
        ...
        
        
        private Boolean isStationsEmpty() {
          return upStationStatus.isEmpty() && downStationStatus.isEmpty();
        }
```

의미있는 이름과 책임을 가지는 메서드로 분리한다면 좀 더 가독성에 좋아 질것 같아요.
물론, 개인적인 관점일수 있습니다 ^^

Reply:
이렇게 수정해보니 훨씬 깔끔해지고 의미가 잘 와닿네요 😃  꿀팁 감사합니다!